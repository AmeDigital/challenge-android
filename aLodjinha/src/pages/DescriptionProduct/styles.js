import { StyleSheet } from 'react-native';

import { RFValue } from 'react-native-responsive-fontsize';

import material from '../../utils/MaterialDesign';

export default StyleSheet.create({
	productDescription: {
		flex: 1,
		alignItems: 'center',
		marginTop: 20
	},
	productImage: {
		height: 200,
		width: 200
	},
	nameDescription: {
		alignSelf: 'stretch',
		fontSize: RFValue(20),
		fontFamily: 'Roboto',
		fontWeight: 'bold',
		color: material.colorDark,
		margin: 15
	},
	containerPrices: {
		borderTopColor: material.colorGreyish,
		borderTopWidth: 1,
		borderBottomColor: material.colorGreyish,
		borderBottomWidth: 1,
		paddingHorizontal: 15,
		paddingVertical: 30,
		flexDirection: 'row',
		alignSelf: 'stretch',
		justifyContent: 'space-between',
		alignItems: 'center',
		height: 40
	},
	priceOf: {
		color: material.colorWhite,
		fontSize: RFValue(12),
		fontWeight: 'bold',
		textDecorationLine: 'line-through'
	},
	priceBy: {
		fontSize: normalize(20),
		color: material.colorTomato,
		fontFamily: 'Roboto',
		fontWeight: 'bold'
	},
	containerDescription: {
		alignSelf: 'stretch'
	},
	title: {
		fontWeight: 'bold',
		fontSize: RFValue(18),
		marginLeft: 15,
		marginTop: 15,
		color: material.colorDark
	},
	description: {
		padding: 17,
		color: material.colorDark,
		letterSpacing: 0.4
	}
});
