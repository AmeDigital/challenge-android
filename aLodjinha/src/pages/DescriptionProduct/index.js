import React, { useEffect, useState } from 'react';
import { useSelector } from 'react-redux';
import { Text, View, StatusBar, Image, ScrollView, Alert } from 'react-native';

import api from '../../services/api';
import FloatingButton from '../../components/FloatingButton';
import { replacePonitForComma } from '../../utils/Utils';

import styles from './styles';
import Loading from '../../components/Loading';

function DescriptionProductScreen({ navigation }) {
	const id = navigation.getParam('id');
	const [ product, setProduct ] = useState(undefined);

	useEffect(
		() => {
			(async function loadProduct() {
				const response = await api.get(`/produto/${id}`);
				setProduct(response.data);
			})();
		},
		[ id ]
	);

	return (
		<View style={{ flex: 1 }}>
			{product !== undefined ? (
				<ScrollView
					onScroll={(event) => {
						if (event.nativeEvent.contentOffset.y > 0) {
							console.log(navigation);
						} else {
							console.log('Top');
						}
					}}
				>
					<View style={styles.productDescription}>
						<StatusBar backgroundColor="#aaaaaa" />
						<Image
							source={{
								uri: product.urlImagem
							}}
							style={styles.productImage}
						/>
						<Text style={styles.nameDescription}>{product.nome}</Text>
						<View style={styles.containerPrices}>
							<Text style={styles.priceOf}>{`De: ${replacePonitForComma(product.precoDe)}`}</Text>
							<Text style={styles.priceBy}>{`Por ${replacePonitForComma(product.precoPor)}`}</Text>
						</View>
						<View style={styles.containerDescription}>
							<Text style={styles.title}>{product.categoria.descricao}</Text>
							<Text style={styles.description}>{product.descricao.replace(/<br\/>+/g, '\n')}</Text>
						</View>
					</View>
				</ScrollView>
			) : (
				<Loading />
			)}
			<FloatingButton
				onPress={() =>
					Alert.alert(null, 'Produto reservado com sucesso', [
						{ text: 'OK', onPress: () => console.log('OK Pressed'), style: '#5e2a84' }
					])}
			/>
		</View>
	);
}

DescriptionProductScreen.navigationOptions = ({ navigation }) => {
	return {
		headerTitle: '',
		headerStyle: {
			backgroundColor: '#fff',
			marginTop: StatusBar.currentHeight,
			elevation: 0
		},
		headerTintColor: '#333'
	};
};

export default DescriptionProductScreen;
