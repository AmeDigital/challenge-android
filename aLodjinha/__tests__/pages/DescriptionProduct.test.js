import React from 'react';

import renderer from 'react-test-renderer';

import DescriptionProduct from '../../src/pages/DescriptionProduct';

it('renders correctly', () => {
	const tree = renderer
		.create(
			<DescriptionProduct
				navigation={{
					getParam: function(param) {
						return 0;
					}
				}}
			/>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
