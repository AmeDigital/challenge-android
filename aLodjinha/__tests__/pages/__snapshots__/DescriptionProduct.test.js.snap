// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#5e2a84"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#5e2a84",
        "borderRadius": 50,
        "bottom": 18,
        "elevation": 4,
        "height": 50,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 30,
        "position": "absolute",
        "right": 18,
        "width": 50,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/images/check_button.png",
        }
      }
      style={
        Object {
          "resizeMode": "contain",
        }
      }
    />
  </View>
</View>
`;
